### PR TITLE
expenses list by userd id

### PR DESCRIPTION
**Pull request by FitaRoberttt**

### CODE REVIEW

File: src/main/java/com/endava/expensesmanager/controller/ExpenseController.java [hunk 1]
```diff
@@ -23,4 +26,10 @@ public ResponseEntity<ExpenseDto> addExpense(@RequestBody @Valid ExpenseDto expe
         expenseService.addExpense(expenseDto);
         return new ResponseEntity<>(expenseDto, HttpStatus.CREATED);
     }
+
+    @GetMapping()
+    public ResponseEntity<List<ExpenseDto>> getExpensesByUserId(@RequestParam Integer userId){
```
Comment (zsoltibv) [inline src/main/java/com/endava/expensesmanager/controller/ExpenseController.java:31]: you can use path variable: /byUser/{userId} according to api contract

**Review by zsoltibv — APPROVED** [review]
ok

File: src/main/java/com/endava/expensesmanager/controller/ExpenseController.java (hunk 1 above)
Comment (zsoltibv) [reply src/main/java/com/endava/expensesmanager/controller/ExpenseController.java:31]: also you have to include as optional params the following: currency, startDate, endDate (see api contract)

Comment (en-dragos-dordea) [reply src/main/java/com/endava/expensesmanager/controller/ExpenseController.java:31]: Currency should not be passed here, this is wrong in the api contract - my bad!

File: src/main/java/com/endava/expensesmanager/service/impl/ExpenseServiceImpl.java [hunk 2]
```diff
@@ -21,4 +24,14 @@ public ExpenseServiceImpl(ExpenseRepository expenseRepository) {
...
         return expenseRepository.save(ExpenseMapper.toExpense(expense));
     }
+
+    @Override
+    public List<ExpenseDto> getExpensesByUserId(Integer userId, LocalDateTime startDate, LocalDateTime endDate){
+
+        return ExpenseMapper.toDtoList(expenseRepository.findAll().stream()
+                .filter(expense -> expense.getUser().getUserId() == userId)
+                .filter(expense -> startDate == null || expense.getExpenseDate().isAfter(startDate) || expense.getExpenseDate().isEqual(startDate))
+                .filter(expense -> endDate == null || expense.getExpenseDate().isBefore(endDate)  || expense.getExpenseDate().isEqual(endDate))
```
Comment (en-dragos-dordea) [inline src/main/java/com/endava/expensesmanager/service/impl/ExpenseServiceImpl.java:34]: This is good, but can be done easier. We can put the stream in a variable and apply filters on date only if we have `startDate` and `endDate`.

File: src/main/java/com/endava/expensesmanager/model/mapper/ExpenseMapper.java [hunk 3]
```diff
@@ -28,4 +30,22 @@ public static Expense toExpense(ExpenseDto expenseDto)
 
         return expense;
     }
+    public static ExpenseDto toDto(Expense expense)
```
Comment (zsoltibv) [inline src/main/java/com/endava/expensesmanager/model/mapper/ExpenseMapper.java:33]: we might need another expense dto that has the expenseId in it, or include it in the current one. We need the ID on frontend for edit and delete actions

File: src/main/java/com/endava/expensesmanager/controller/ExpenseController.java [hunk 4]
```diff
@@ -23,4 +27,11 @@ public ResponseEntity<ExpenseDto> addExpense(@RequestBody @Valid ExpenseDto expe
         expenseService.addExpense(expenseDto);
         return new ResponseEntity<>(expenseDto, HttpStatus.CREATED);
     }
+
+    @GetMapping("byUser/{userId}")
+    public ResponseEntity<List<ExpenseDto>> getExpensesByUserId(@PathVariable Integer userId, @RequestParam(required = false) LocalDateTime startDate, @RequestParam(required = false) LocalDateTime endDate){
```
Comment (en-dragos-dordea) [inline src/main/java/com/endava/expensesmanager/controller/ExpenseController.java:32]: `startDate` and `endDate` should be of type `LocalDate`, as we are not going to work with time.

**Review by en-dragos-dordea — CHANGES_REQUESTED** [review]

**Review by en-dragos-dordea — APPROVED** [review]

**Pull request merged**